examples/adapt/image_segmentation/README.md

```diff
@@ -0,0 +1,206 @@
+# SegNeXt on cityscapes dataset using SSF 
+
+## Overview 
+
+This guide provides a walkthrough of applying SegNeXt for instance segmentation on the cityscapes dataset using SSF (Scaling and Shifting the deep Features). SSF enables parameter efficient fine-tuning by proposing that performace simillar to full fine-tuning can be achieved by only scaling and shifting the features of a deep neural network. 
+
+## Table on Contents
+ - [Introduction](#introduction)
+ - [Requirements](#requirements)
+ - [Installation](#installation)
+ - [Dataset](#dataset)
+ - [Configuration](#configuration)
+ - [Adapting the model](#adapting-the-model)
+ - [Conclusion](#conclusion)
+ 
+
+## Introduction
+
+In this example we will be finetuning SegNeXt large model on a city scapes dataset uisng a PEFT technique called SSF (Scale and Shift deep Features). SSF uses two linear layers to learn the scale and shift factors for deep features and hence uses very few trainable parameters. 
+
+## Requirements
+
+Before you begin, ensure that you have the following:
+- A GPU-enabled environment with CUDA support.
+- The Nyuntam repository cloned and set up as per the [Installation Guide](#installation).
+- Docker
+
+## Installation
+
+### Step 1: Clone the Nyuntam Repository
+
+Clone the repository and navigate to the `nyuntam` directory:
+```bash
+$ git clone https://github.com/nyunAI/nyuntam.git
+$ cd nyuntam
+```
+
+### Step 2: Set Up the workspace
+
+To setup the environment use the following command(s),
+
+```bash
+pip install git+https://github.com/nyunAI/nyunzero-cli.git
+nyun init {WORKSPACE_PATH} -e adapt
+```
+
+## Dataset 
+
+For this experiment we will be using the [CityScapes](https://www.kaggle.com/datasets/ipythonx/stanford-background-dataset) dataset. The original folder structure of the dataset loks like :  
+```bash
+iccv09Data
+└── images
+|   └── image1.jpg
+|   └── image2.jpg
+|   └── image3.jpg
+    ...
+└── seg_maps
+|   └── image1.layers.txt
+|   └── image1.regions.txt
+|   └── image1.surfaces.txt
+└── ann_file.txt
+└── horizons.txt
+```
+
+The text files contain the segmentation maps in digits. In this dataset we will only be using the "regions" as the segmentation maps. These are the text files which are named as "image1.regions.txt". These text files are converted to ".png" images.  
+The new data folder structure is formatted in the folowing way: 
+
+```bash
+iccv09Data
+└── images
+|   └── image1.jpg
+|   └── image2.jpg
+|   └── image3.jpg
+    ...
+└── seg_maps
+    └── image1.png
+    └── image2.png
+    └── image3.png
+    ...
+└──train_ann.txt (text file containing the names of training images without the extension)
+└──val_ann.txt (text file containing the names of validation images without the extension)
+```
+
+## Configuration
+
+The following YAML file is used for setting up the experiment : 
+
+```yaml
+
+#DATASET
+JOB_SERVICE: Adapt
+CUSTOM_DATASET_PATH : "/custom_data/iccv09Data" 
```

Comment:
is this path correct? custom_data as an absolute path seems incorrect.

Reply:
all paths in the cli flow are absolute paths. The "custom_data" path while creating the Nyun-workspace is mounted to "/custom_data" in the docker hence the path.

Reply:
Adapt -https://github.com/nyunAI/nyuntam_adapt/blob/228e5ec7394dc03d2d88fdda4ed3c85295310fd3/core/base_task.py#L72

Text-Gen - 
https://github.com/nyunAI/nyuntam-text-generation/blob/9542023ef0836e4346c45eecaad83711c71efadc/core/dataset.py#L60

Vision -
https://github.com/nyunAI/nyuntam-vision/blob/edcc1ff5e9061351efa2b974b0591b4d05df973b/core/models/__init__.py#L34

The custom paths are directly checked for existence. Without prefixing any parent custom_path (i.e. not considering as relative path) and hence the above.